src/app/system/manage-surveys/create-survey/create-survey.component.ts

```diff
@@ -0,0 +1,151 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, FormArray, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { SystemService } from '../../system.service';
+
+/**
+ * Create survey component.
+ */
+@Component({
+  selector: 'mifosx-create-survey',
+  templateUrl: './create-survey.component.html',
+  styleUrls: ['./create-survey.component.scss']
+})
+export class CreateSurveyComponent implements OnInit {
+
+  /** Survey form. */
+  surveyForm: FormGroup;
+
+  /**
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService System Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private systemService: SystemService,
+              private route: ActivatedRoute,
+              private router: Router) { }
+
+  /**
+   * Creates the survey form.
+   */
+  ngOnInit() {
+    this.createSurveyForm();
+  }
+
+  /**
+   * Creates the survey form.
+   */
+  createSurveyForm() {
+    this.surveyForm = this.formBuilder.group({
+      'key': ['', Validators.required],
+      'name': ['', Validators.required],
+      'countryCode': ['', [Validators.required, Validators.pattern('^\\s*([A-Z]{2})?\\s*$')]],
+      'description': [''],
+      'questionDatas': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Gets the questions form array.
+   * @returns {FormArray} Questions form array.
+   */
+  get questionDatas(): FormArray {
```

Comment:
'questionDatas' is awkward, but I have named it as so rather than just 'question' or 'questionData' because I would like it to match the variable name from the API, which would make it easier for other people working on this in the future to understand it better.